src/docs/development/platform-integration/platform-channels.md

```diff
@@ -637,6 +636,52 @@ If using the iOS Simulator,
 note that it does not support battery APIs,
 and the app displays 'battery info unavailable'.
 
+## Typesafe platform channels via Pigeon {#pigeon}
+
+The previous example uses `MethodChannel` to communicate between the host and
+client which isn't typesafe.  Calling and receiving messages depends on the host
+and client declaring the same arguments and datatypes in order for messages to
+work.  The [Pigeon][] package can be used as an alternative to `MethodChannel`
+to generate code that sends messages in a typesafe manner.
+
+With [Pigeon][] the messaging protocol is defined in a subset of Dart which then
+generates messaging code for Android or iOS.  A more complete example and more
+information can be found on the [Pigeon pub.dev page][];
+
+### Pigeon example
+
+**Pigeon file:**
+
+```dart
+import 'package:pigeon/pigeon.dart';
+
+class SearchRequest {
+  String query;
+}
+
+class SearchReply {
+  String result;
+}
+
+@HostApi()
+abstract class Api {
+  SearchReply search(SearchRequest request);
+}
+```
+
+**Dart usage:**
+
+```dart
+import 'generated_pigeon.dart'
+
+void onClick() async {
+  SearchRequest request = SearchRequest()..query = 'test';
+  Api api = Api();
+  SearchReply reply = await api.search(request);
+  print('reply: ${reply.result}');
+}
+```
```

Comment:
We should mention that it's 2 directional too.

Reply:
done, added list of benefits